InClassAssignment2-LenaL.Rmd

```diff
@@ -0,0 +1,175 @@
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+V.STD2<- dplyr::select(D1, stid, s.major)
+V.STD2<-dplyr::count(V.STD2, stid, s.major)
+V.STD2$group <- "student"
+
+
+V.TCH2<- dplyr::select(D1, tid, t.expertise)
+V.TCH2<-dplyr::count(V.TCH2, tid, t.expertise)
+V.TCH2$group <- "teacher"
+
+names(V.TCH2) <- c("id",  "topic", "count", "group")
+names(V.STD2) <- c("id",  "topic", "count","group")
+
+VERTEX2 <- dplyr::bind_rows(V.TCH2, V.STD2)
+
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX2)
+
+plot(g2,layout=layout.fruchterman.reingold, vertex.size=VERTEX2$count, edge.width=EDGE$count)
+```
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+```{r}
+#load the library we need
+library(dplyr)
+library(tidyr)
+ourClass<-read.csv("HUDK4050-2018-classes.csv")
+D3<-select(ourClass, Q8, Q9, Q1, Q3, Q4, Q5)
+D3<-D3 %>% unite(student,Q8,Q9,sep=' ')
+D3 <- D3[-c(1,2,39),]
+
+# formalize the data
+D3[D3=="HUDK4050"]<-"HUDK 4050"
+D3[D3=="HUDK5035"]<-"HUDK 5035"
+D3[D3=="HUDK4011"]<-"HUDK 4011"
+D3[D3=="HUDM4125"]<-"HUDM 4125"
+D3[D3=="MSTU5003"]<-"MSTU 5003"
+D3[D3=="MSTU4083"]<-"MSTU 4083"
+D3[D3=="MSTU4039"]<-"MSTU 4039"
+
+#count the number of courses by gather and spread
+D4<-D3%>% gather(key=var1,value = var2,...=2:5,na.rm = TRUE)
+D5<-D4%>%count(student,var2)
+D6<-filter(D5, var2!=""&var2!="HUDK 4050")
+D7<-D6%>%spread(key=var2, value = n)
+
+#remove NA
+D7[is.na(D7)]<-0
+
+# convert df to matrices
+D7_matrix<-as.matrix(D7[-1])
+rownames(D7_matrix)<-D7$student
+D8_matrix<-t(D7_matrix)
+ad_matrix<-D7_matrix %*%D8_matrix
+diag(ad_matrix)<-0
+
+#plot the graph 
+library(igraph)
+plot(graph_from_adjacency_matrix(ad_matrix, mode="undirected", weighted=NULL, diag=TRUE), layout= layout.fruchterman.reingold)
+```
+
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and degree. **Who is the most central person in the network?**
```

Comment:
You've generated centrality and betweeness metrics, but what do they tell you? The slides from class should help with this.